nd4j/src/main/scala/nexus/impl/nd4j/ND4JBackend.scala

```diff
@@ -0,0 +1,168 @@
+package nexus.impl.nd4j
+
+import nexus.algebra._
+import nexus.algebra.typelevel.util._
+import org.nd4j.linalg.factory.Nd4j
+import org.nd4j.linalg.ops.transforms.Transforms
+
+import scala.util._
+
+
+/**
+  * @author Andrey Romanov
+  */
+object ND4JBackendTensor extends IsRealTensorK[FloatTensor, Float]{
+  type H = UntypedFloatTensor
+  val H = UntypedND4JBackendTensor
+
+  val R = H.R
+
+  def newTensor[A](shape: Array[Int]) =
+    FloatTensor.fromFlatArray[A](Array.ofDim[Float](ShapeUtils.product(shape)), shape)
+
+  def newGaussianTensor[A](μ: Double, σ2: Double, shape: Array[Int]) = {
+    val r = new Random()
+    val σ = math.sqrt(σ2)
+    FloatTensor.fromFlatArray(Array.fill(ShapeUtils.product(shape))(((r.nextGaussian() - μ) * σ).toFloat), shape)
+  }
+
+  def get(x: FloatTensor[_], is: Array[Int]): Float = {
+    x.handle.getFloat(x.index(is))
+  }
+  def set(x: FloatTensor[_], is: Array[Int], v: Float) = x.update(is: _*)(v)
+  def zeroBy[A](x: FloatTensor[A]) = newTensor(x.shape)
+
+  def untype(x: FloatTensor[_]) = x
+  def typeWith[A](x: UntypedFloatTensor) = x.typeWith[A]
+
+  def fromDouble(d: Double) = d.toFloat
+  def fromFloat(f: Float) = f
+}
+
+
+object UntypedND4JBackendTensor extends IsUntypedRealTensor[UntypedFloatTensor, Float] {
```

Comment:
There's no need to create the wrapper class `UntypedFloatTensor`. Just using `INDArray` would do.
I prefer that it is directly written as `object UntypedND4JTensor extends IsUntypedRealTensor[INDArray, Float]`.

P.S. Does ND4J supports elements other than Float? Does it support things like `Int`?

Reply:
Good point. 
Actually ND4J only supports either `double` or `float` types and this is set globally
```java
Nd4j.setDataType(DataBuffer.Type.DOUBLE)
``` 
and all NDArrays should have the same data type.

And it seems that [support for multiple data types](https://github.com/deeplearning4j/nd4j/issues/1690)  will not be in the near future.

Reply:
Hmm. This creates problems for us since Nexus has `IsIntTensorK` -- tailored from ops like `ArgMax` which returns `Int` tensors. I'm wondering how `Nd4j`/`Dl4j` deals with these

Reply:
`Nd4j` returns a `Float` tensor with `1.0`and `0.0` values as a result of `argmax` operation :\